### DISCUSSION TITLE
Schema awareness?

### DISCUSSION THREAD

**Discussion by jaytoday** [outerbase/ezql#21] [General]
In the example in the README, a SQL query is generated:

```

  ez.prompt(phrase, Outerbase.Prompt.sql).then((sqlText) => console.log(sqlText))
  /* "SELECT answer FROM ultimate_question;" */
  ```

How does the agent know to use table `ultimate_question` and column `answer`? Are these just guesses or has it omitted a step where a database schema is provided for context?

**Comment by Brayden** [comment]
@jaytoday Great question. When using it with the Outerbase API source token we do also pass in the database schema under the hood without other users having to be worried about doing that lift themselves. So we'll pass in both the natural language request such as:

```
Show me all users
```

And on the backend side of our service we'll pass in what the schema of the database kind of looks like.

```
Users(id, first_name, last_name)
```

And with all of that context it will construct the appropriate SQL statement that can be executed. The TLDR; of this is that if you sign up with an Outerbase account and connect your database to it, we do the rest.
